applications/Shopbring.md

```diff
@@ -0,0 +1,320 @@
+
+#### Credit Incentive Mechanism
+
+For the healthy development of Shopbring, we design a credit value growth model and native token incentive algorithm. The core meaning is as follows:
+
+* The behavior that is favorable to the development of Shopbring will increase the credit value, and the behavior that is unfavorable to the development of Shopbring will reduce the credit value.
+* Open and transparent credit data can help users choose trading partners.
+* In Polkadot ecosystem, it provides credit data reference for other parachains.
+
+In order to encourage the Shopbring users to increase the credit value, the system will determine the number of airdrop tokens according to the average credit rating of the overall users in each cycle. The higher the average credit value, the greater the total number of airdrop tokens. The number of airdrop tokens obtained by users = the number of single credit value airdrop tokens * the user credit value.
+
+#### Arbitrator
+
+Normal user registration arbitrators need to freeze a certain native token as an application condition. The `council` will review the applicant's information, and more than half of the members will vote to become an arbitrator.
+
+The arbitrator’s obligation is that when a common user has a dispute during the commissioned shopping, either party will appeal to the system. The arbitrator can choose the case to intervene in the dispute. During the waiting period, up to `M` arbitrators can sign up to participate in the dispute. After the waiting period ends, the system will randomly select `N`($N < M$, and `N` is an odd number) arbitrators to actually participate in the dispute. If there are less than `N` arbitrators actually participating, the `council` will participate in dispute arbitration. After the arbitration is completed, the `treasury` will pay wages to the participating arbitrators.
+
+#### Inspector
+
+Shopbring introduces the role of inspector to verify whether the shopping process actually happened. This role provides trusted services to verify shopping receipts(like invoices). For example, in China, most e-commerce platforms provide the right of applying for electronic invoices issued by the State Administration of Taxation. By verifying the authenticity of the electronic invoice, it can be determined whether the consumption process has occurred. Shopbing has the following design requirements for inspectors:
+
+* The user submits the application for registering inspector, then waits for the council to review. When more than half of the members of the council vote for, the review is passed.
+* The inspector needs to provide an API service to verify the shopping receipt. Shopbring's off-chain worker will submit the unverified shopping receipt to the inspector's API service, and finally the inspector will submit the verification result to the chain.
```

Comment:
**We define the structure of shopping receipt like this：**

| Parameter      | Type     | Description                                                                                   |
|----------------|----------|-----------------------------------------------------------------------------------------------|
| area           | u32      | Country or area code.                                                                         |
| receipt_type   | u8       | Receipt type: 0: e-invoice, 1: physical bill, need to be photographed and uploaded.           |
| payload        | Vec\<u8> | E-invoice information is saved in json format according to different national standards.      |
| receipt_attach | Vec\<u8> | The link to the physical bill attachment requires manual verification.                        |
| receipt_hash   | Vec\<u8> | The receipt's data will be encoded using SCALE-codec and finally be hashed using blake2b-256. |
| order_hash     | Vec\<u8> | Order hash, used to associate original order detail.                                         |

1. The registered inspector can bind his verification API on-chain.
2. Because there may be many types of receipt formats, we store the detail of the receipt off-chain, and after encoding the data, submit the `receipt hash` to the chain for storage.
3. The verification system of the inspector needs to connect to the off-chain order system, and send the `receipt hash` to it to fetch the detail of the receipt. The verification system completes verification of the receipt data and records the results.
4. The off-chain worker sends the `receipt hash` to the verification API, fetchs the verification result, and submits the transaction signed by the inspector account to the chain.